File: src/dataset/Java/bob_exercism-ThomasZumsteg/openAI/ChatGPT4_Bob_v2.java
public class Bob {
    public String hey(String greeting) {
        if (greeting == null || greeting.isBlank()) {
            return "Fine. Be that way!";
        }
        greeting = greeting.trim();
        if (isShouting(greeting)) {
            return "Whoa, chill out!";
        }
        if (greeting.endsWith("?")) {
            return "Sure.";
        }
        return "Whatever.";
    }

    private static boolean isShouting(String greeting) {
        boolean hasUppercase = false;
        for (int i = 0; i < greeting.length(); i++) {
            char c = greeting.charAt(i);
            if (Character.isLowerCase(c)) {
                return false;
            }
            if (Character.isUpperCase(c)) {
                hasUppercase = true;
            }
        }
        return hasUppercase;
    }
}

File: src/dataset/javascript/bob_exercism-ThomasZumsteg/openAI/ChatGPT4_bob_v2.js
class Bob {
  hey(input) {
    input = input.trim();
    if (input === '') return "Fine. Be that way!";
    const isShouting = input === input.toUpperCase() && /[A-Z]/.test(input);
    const isQuestion = input.endsWith('?');
    if (isShouting) return "Whoa, chill out!";
    if (isQuestion) return "Sure.";
    return "Whatever.";
  }
}

module.exports = Bob;